lib/client/db/dbcmd/dbcmd_test.go

```diff
@@ -0,0 +1,534 @@
+			cmd: []string{mssqlBin,
+				"-S", "localhost,12345",
+				"-U", "myUser",
+				"-P", fixtures.UUID,
+				"-d", "mydb",
+			},
+			wantErr: false,
+		},
+		{
+			name:       "redis-cli",
+			dbProtocol: defaults.ProtocolRedis,
+			execer:     &fakeExec{},
+			cmd: []string{"redis-cli",
+				"-h", "localhost",
+				"-p", "12345",
+				"--tls",
+				"--key", "/tmp/keys/example.com/bob",
+				"--cert", "/tmp/keys/example.com/bob-db/db.example.com/mysql-x509.pem"},
+			wantErr: false,
+		},
+		{
+			name:         "redis-cli with db",
+			dbProtocol:   defaults.ProtocolRedis,
+			databaseName: "2",
+			execer:       &fakeExec{},
+			cmd: []string{"redis-cli",
+				"-h", "localhost",
+				"-p", "12345",
+				"--tls",
+				"--key", "/tmp/keys/example.com/bob",
+				"--cert", "/tmp/keys/example.com/bob-db/db.example.com/mysql-x509.pem",
+				"-n", "2"},
+			wantErr: false,
+		},
+		{
+			name:       "redis-cli no TLS",
+			dbProtocol: defaults.ProtocolRedis,
+			execer:     &fakeExec{},
+			noTLS:      true,
+			cmd: []string{"redis-cli",
+				"-h", "localhost",
+				"-p", "12345"},
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			database := &tlsca.RouteToDatabase{
+				Protocol:    tt.dbProtocol,
+				Database:    tt.databaseName,
+				Username:    "myUser",
+				ServiceName: "mysql",
+			}
+
+			opts := []ConnectCommandFunc{
+				WithLocalProxy("localhost", 12345, ""),
+			}
+			if tt.noTLS {
+				opts = append(opts, WithNoTLS())
+			}
+
+			c := NewCmdBuilder(tc, profile, database, "root", opts...)
+			c.uid = utils.NewFakeUID()
+			c.exe = tt.execer
+			got, err := c.GetConnectCommand()
+			if tt.wantErr {
+				if err == nil {
+					t.Errorf("getConnectCommand() should return an error, but it didn't")
+				}
+				return
+			}
+
+			require.NoError(t, err)
+			require.Equal(t, tt.cmd, got.Args)
+		})
+	}
+}
+
+// // Similar to fakeExec, but always returns a Command with an absolute path.
+// type fakeAbsolutePathExec struct {
+// 	// execOutput maps binary name and output that should be returned on RunCommand().
+// 	// Map is also being used to check if a binary exist. Command line args are not supported.
+// 	execOutput map[string][]byte
+// }
+
+// func (f fakeAbsolutePathExec) RunCommand(cmd string, _ ...string) ([]byte, error) {
+// 	out, found := f.execOutput[cmd]
+// 	if !found {
+// 		return nil, errors.New("binary not found")
+// 	}
+
+// 	return out, nil
+// }
+
+// func (f fakeAbsolutePathExec) LookPath(path string) (string, error) {
+// 	if _, found := f.execOutput[path]; found {
+// 		return "", nil
+// 	}
+// 	return "", trace.NotFound("not found")
+// }
+
+// func (f fakeAbsolutePathExec) Command(name string, arg ...string) *exec.Cmd {
+// 	path, err := os.Getwd()
+// 	if err != nil {
+// 		panic(err)
+// 	}
+
+// 	absolutePath := filepath.Join(path, name)
+// 	cmd := exec.Command(absolutePath, arg...)
+
+// 	return cmd
+// }
```

Comment:
Shouldn't this be removed?

Reply:
It absolutely should, thanks for pointing it out.